### PR TITLE
Adding windows wheel builds

### PR DESCRIPTION
**Pull request by atalman**
Adding windows wheel builds

### CODE REVIEW

**Comment by pmeier** [comment]
The pre-script run has multiple failures that don't fail the run for some reason

- https://github.com/pytorch/vision/actions/runs/4853188938/jobs/8649078064#step:8:28
- https://github.com/pytorch/vision/actions/runs/4853188938/jobs/8649078064#step:8:31

File: .github/workflows/build-wheels-windows.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      ref: ""
+      test-infra-repository: pytorch/test-infra
+      test-infra-ref: main
+      build-matrix: ${{ needs.generate-matrix.outputs.matrix }}
+      pre-script: ${{ matrix.pre-script }}
+      env-script: ${{ matrix.env-script }}
+      post-script: ${{ matrix.post-script }}
+      package-name: ${{ matrix.package-name }}
+      smoke-test-script: ${{ matrix.smoke-test-script }}
+      trigger-event: ${{ github.event_name }}
```
Comment (osalpekar) [inline .github/workflows/build-wheels-windows.yml:49]: Same as comment on the audio PR: Let's run these in test mode for ~1 week prior to uploading from Nova. You can set this field to "dev" to do so.

**Comment by atalman** [comment]
> The pre-script run has multiple failures that don't fail the run for some reason
> 
> * https://github.com/pytorch/vision/actions/runs/4853188938/jobs/8649078064#step:8:28
> * https://github.com/pytorch/vision/actions/runs/4853188938/jobs/8649078064#step:8:31

Done, isolated part of code that does not work on windows

**Review by pmeier — APPROVED** [review]
Stamping to unblock. Wondering if we should run the pre and post scripts with `set -e` to avoid this in the future?

**Pull request merged**